src/v/kafka/server/connection_context.h

```diff
@@ -349,6 +380,7 @@ class connection_context final
     sequence_id _next_response;
     sequence_id _seq_idx;
     map_t _responses;
+    ssx::sharded_abort_source _as;
```

Comment:
Does having a `ss::sharded` object per connection context make the connection context/connection creation quite expensive as it forces lots of cross cpu talk? 

Not sure it matters but just wondering.

Reply:
To some extent, yes. 

Cross-core calls occur:
* `start()`
* `stop()`
* `request_abort()`

The trade-off is that every core has a local `abort_source` to query for:
* `abort_requested()`
* `check()`
* `subscribe()`

It might be possible to create the per-core `abort_source` lazily:
* Futurize the call to `local()`
* If the local one is constructed, return it
* Cross-core call to request to construct it on the owning core
* Maybe lock a mutex
* Cross-core call (back to local) to actually construct it

It's a fair bit of complexity, and makes the interface more awkward.